### ISSUE TITLE
[Feature Request] Import raw data from taskwarrior

### ISSUE THREAD

**Issue by KevCui** [soywod/kronos.vim#11]
Currently, I'm using taskwarrior for managing my daily tasks. I'd like to start using kronos. Is there a plan to add the feature which can import the taskwarrior raw data to kronos `.database`? It would make life easier for doing transition. Thanks!

**Comment by soywod** [comment]
This is the raw data you were referring to https://taskwarrior.org/docs/design/task.html? I guess it's possible, but you will loose some data, because Kronos is not as complete as Taskwarrior. Could you share your export (so I can give a try)?

**Comment by KevCui** [comment]
Yes, you're right about the data. No worry if it's not compatible with taskwarrior data, the worst case I can replicate tasks in Kronos manually or using a script.

Here is a sample data I created using taskwarrior. Let me know if you need my support on testing or providing more info.

File: **pending.data**
```
[description:"test task" due:"1558216799" entry:"1557674029" modified:"1557674029" status:"pending" tags:"work" uuid:"c70ae202-9070-4479-a337-fa9df6264422"]
[description:"blog idea collection" due:"1557698400" entry:"1557674045" modified:"1557674045" status:"pending" tags:"blog" uuid:"2e0e612a-c34d-4d19-b8a8-b219b81c9999"]
[description:"visit xxx" due:"1559289600" entry:"1557674098" modified:"1557674098" status:"pending" tags:"personal" uuid:"8f1a2c86-32cc-478c-9d4f-0525dee3fd28"]
[description:"watch GoT" due:"1557698400" entry:"1557674282" mask:"-" modified:"1557674315" recur:"weekly" status:"recurring" tags:"personal" uuid:"3c9ad235-7c27-4564-af77-b2c03a6e532c"]
[description:"watch GoT" due:"1557698400" entry:"1557674283" imask:"0" modified:"1557674313" parent:"3c9ad235-7c27-4564-af77-b2c03a6e532c" recur:"weekly" status:"pending" tags:"personal" uuid:"d56c08c4-24b5-4cd1-b2f8-46c45bf5321f"]
[description:"something else" entry:"1557674295" modified:"1557674295" status:"pending" uuid:"023f2695-bed8-4def-ac97-6c93d0132e82"]
[description:"shopping" due:"1557784800" entry:"1557674361" modified:"1557674361" status:"pending" tags:"personal" uuid:"3c380186-bb0d-4fd5-98ad-37579f86f44f"]
```

File: **backlog.data**
```
{"description":"test task","due":"20190518T215959Z","entry":"20190512T151349Z","modified":"20190512T151349Z","status":"pending","tags":["work"],"uuid":"c70ae202-9070-4479-a337-fa9df6264422"}
{"description":"blog idea collection","due":"20190512T220000Z","entry":"20190512T151405Z","modified":"20190512T151405Z","status":"pending","tags":["blog"],"uuid":"2e0e612a-c34d-4d19-b8a8-b219b81c9999"}
{"description":"visit xxx","due":"20190531T080000Z","entry":"20190512T151458Z","modified":"20190512T151458Z","status":"pending","tags":["personal"],"uuid":"8f1a2c86-32cc-478c-9d4f-0525dee3fd28"}
{"description":"shopping","due":"20190512T220000Z","entry":"20190512T151802Z","modified":"20190512T151802Z","recur":"weekly","status":"recurring","tags":["personal"],"uuid":"3c9ad235-7c27-4564-af77-b2c03a6e532c"}
{"description":"shopping","due":"20190512T220000Z","entry":"20190512T151803Z","imask":0,"modified":"20190512T151802Z","parent":"3c9ad235-7c27-4564-af77-b2c03a6e532c","recur":"weekly","status":"pending","tags":["personal"],"uuid":"d56c08c4-24b5-4cd1-b2f8-46c45bf5321f"}
{"description":"shopping","due":"20190512T220000Z","entry":"20190512T151802Z","mask":"-","modified":"20190512T151803Z","recur":"weekly","status":"recurring","tags":["personal"],"uuid":"3c9ad235-7c27-4564-af77-b2c03a6e532c"}
{"description":"something else","entry":"20190512T151815Z","modified":"20190512T151815Z","status":"pending","uuid":"023f2695-bed8-4def-ac97-6c93d0132e82"}
{"description":"watch GoT","due":"20190512T220000Z","entry":"20190512T151803Z","imask":0,"modified":"20190512T151833Z","parent":"3c9ad235-7c27-4564-af77-b2c03a6e532c","recur":"weekly","status":"pending","tags":["personal"],"uuid":"d56c08c4-24b5-4cd1-b2f8-46c45bf5321f"}
{"description":"watch GoT","due":"20190512T220000Z","entry":"20190512T151802Z","mask":"-","modified":"20190512T151835Z","recur":"weekly","status":"recurring","tags":["personal"],"uuid":"3c9ad235-7c27-4564-af77-b2c03a6e532c"}
{"description":"shopping","due":"20190513T220000Z","entry":"20190512T151921Z","modified":"20190512T151921Z","status":"pending","tags":["personal"],"uuid":"3c380186-bb0d-4fd5-98ad-37579f86f44f"}
```
![taskwarrior](https://user-images.githubusercontent.com/14293537/57584576-4d995680-74dd-11e9-9df4-d3f48e5c23d0.jpg)

**Comment by soywod** [comment]
I pushed something, here the result:

![image](https://user-images.githubusercontent.com/10437171/57585633-fdc18c00-74ea-11e9-8ebe-a1bb7e80f444.png)

I pushed on `import-from-taskwarrior` branch, so you need to adapt your vim conf. If you use `vim-plug`, you could do:

```vim
Plug 'soywod/kronos.vim', {'branch': 'import-from-taskwarrior'}
```

You will see a `import.py` at the root folder of the plugin. Usage: `./import.py <provider> <filepath>`, where `provider` is `taskwarrior`, and `filepath` the path to your backup. Let me know.

**Comment by KevCui** [comment]
Thanks for creating the `import.py` script. There are some small issues when importing data and I tried to fix them in pull request #13 :smile:

**Comment by soywod** [comment]
#14 merged :tada:
